### PR TITLE
feat(fe): add code editor component

### LINKED ISSUE #190: [Feature] Problem(Molecule) -  CodeEditor
### 🎨 Component Image

image

### ✏️ Describe component

CodeMirror를 이용한 코드 에디터입니다.

VSCode의 전신인 Monaco와 고민을 많이 했는데, 결정적으로 모바일 지원을 위해 CodeMirror를 택했습니다. (Monaco는 미지원)
최근 업데이트된 v6을 사용할 예정입니다.
https://codemirror.net/

언어별 highlight, 자동 indentation, 문법 오류 표시 등 코드 에디터 기능을 갖춘 component입니다.
(자동 완성(IntelliSense)은 좀 더 알아보고 구현할 지 정하겠습니다!)

### ☁️ Props

WIP

### 🔎 Using Page

(당연히) Problem 페이지

제출 결과 등 readonly로 코드 보여주는 페이지는 별도의 component를 사용할 지 고민 중입니다.
정해지면 이 부분도 수정하겠습니다!

### 🏷 Required Component

_No response_

### PR DESCRIPTION
**Pull request by dotoleeoak**
### Description

Close #190 

CodeMirror(v6) 기반의 코드 편집기입니다.

지금까지 언어별(C/C++, Python, Javascript, Java) syntax highlighting과 indentation, bracket closing 등을 지원합니다.
Linting(문법 검사)과 intellisense도 지원하고 싶었지만, 해당 부분은 더 알아보고 나중에 PR 올리는 게 좋을 것 같습니다!

image

---

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md)
- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#pr-and-branch) and follow the [Commit Convention](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#commit-convention)
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: frontend/src/common/components/Organism/CodeEditor.vue [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const extensions = computed(() => [
+  keymap.of([...defaultKeymap, ...historyKeymap, indentWithTab]),
+  oneDark,
+  font,
+  languageExtensions[props.lang],
+  history(),
+  lineNumbers(),
+  highlightActiveLine(),
+  drawSelection(),
+  closeBrackets(),
```
Comment (jimin9038) [inline frontend/src/common/components/Organism/CodeEditor.vue:67]: Codemirror reference에는 ( ) 를 쳤을 때 ( ) 가 출력되어야 할 것 같은데 (참고: When closing a bracket directly in front of a closing bracket inserted by the extension, the cursor moves over that bracket. ) 이상하게 ( ) )이 출력되네요..? 왜일까요..

File: frontend/src/common/components/Organism/CodeEditor.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      if (tr.changes.empty) return
+      const v = view.value.state.doc.toString()
+      emit('update:modelValue', v)
+    }
+  })
+})
+</script>
+
+<template>
+  <div ref="editor" class="h-80 w-full" />
```
[the remark this answers is not in the document]
Comment (dotoleeoak) [reply frontend/src/common/components/Organism/CodeEditor.vue]: 그게 좋겠네요. 적용했습니다!

File: frontend/src/common/components/Organism/CodeEditor.vue (hunk 1 above)
Comment (dotoleeoak) [reply frontend/src/common/components/Organism/CodeEditor.vue:67]: 해결했습니다!

**Review by jimin9038 — APPROVED** [review]
수고하셨습니다!

**Pull request merged**